src/cpu/aarch64/acl_winograd_convolution.cpp

```diff
@@ -14,29 +14,129 @@
 * limitations under the License.
 *******************************************************************************/
 
-#include "cpu/aarch64/acl_winograd_convolution.hpp"
+#include "acl_winograd_convolution.hpp"
+#include "common/memory_tracking.hpp"
+#include "common/utils.hpp"
 
 namespace dnnl {
 namespace impl {
 namespace cpu {
 namespace aarch64 {
 
+namespace {
+// Keys are anonymous. So deduce the type automagically.
+using conv_key_t = decltype(memory_tracking::names::key_gemm_tmp_buffer);
+
+// Map: [slot , key]
+const std::map<int, conv_key_t> wino_conv_keys
+        = {{0, conv_key_t::key_gemm_asm_tmp_buffer},
+                {1, conv_key_t::key_gemm_pretranspose_b},
+                {2, conv_key_t::key_gemm_pretranspose},
+                {3, conv_key_t::key_gemm_interleaved_lhs},
+                {4, conv_key_t::key_gemm_pretransposed_rhs},
+                {5, conv_key_t::key_gemm_transposed_1xwrhs},
+                {6, conv_key_t::key_gemm_tmp_buffer},
+                {7, conv_key_t::key_conv_permuted_outputs},
+                {8, conv_key_t::key_conv_permuted_inputs},
+                {9, conv_key_t::key_wino_workspace},
+                {10, conv_key_t::key_wino_transformed_weights},
+                {11, conv_key_t::key_conv_permuted_weights}};
+} // namespace
+
+status_t acl_wino_convolution_fwd_t::pd_t::init(engine_t *engine) {
+    using namespace data_type;
+    const bool is_fp16_ok = expect_data_types(f16, f16, f16, f16, undef)
+            && attr()->has_default_values(
+                    primitive_attr_t::skip_mask_t::post_ops, f16);
+    const bool is_fp32_ok = expect_data_types(f32, f32, f32, f32, undef)
+            && attr()->has_default_values(
+                    primitive_attr_t::skip_mask_t::post_ops, f32);
+    bool ok = is_fwd()
+            && utils::one_of(desc()->alg_kind, alg_kind::convolution_auto,
+                    alg_kind::convolution_winograd)
+            && utils::one_of(true, is_fp16_ok, is_fp32_ok)
+            && !has_zero_dim_memory();
+
+    ok = ok && DNNL_CPU_THREADING_RUNTIME != DNNL_RUNTIME_THREADPOOL;
+    if (!ok) return status::unimplemented;
+
+    CHECK(init_conf());
+
+    set_default_alg_kind(alg_kind::convolution_winograd);
+
+    Op conv;
+    conv.configure(&acp_.src_tensor_info, &acp_.wei_tensor_info,
+            acp_.with_bias ? &acp_.bia_tensor_info : nullptr,
+            &acp_.dst_tensor_info, acp_.padstride_info, acp_.act_info,
+            true); // to support 5x5, 7x7 filter shapes in addition to 3x3
+
+    auto scratchpad = scratchpad_registry().registrar();
+    const auto aux_mem = conv.workspace();
+    return init_scratchpad(conv, scratchpad, wino_conv_keys, engine, post_ops,
+            attr_.post_ops_, acp_.act_info, acp_.use_dst_acc_for_sum, dst_md_);
+}
+
+status_t acl_wino_convolution_fwd_t::create_resource(
+        engine_t *engine, resource_mapper_t &mapper) const {
+    CHECK(pd()->post_ops.create_resource(engine, mapper));
```

Comment:
QQ: though the main op seems to get rid of a resource dependency, post-ops are still using the `mapper`, correct?

Reply:
Yep, you are right. We are working internally to sort that out. A few more patches still to go.